### ISSUE TITLE
Advanced tricks required for casual mode

### ISSUE THREAD

**Issue by mkfanha** [tewtal/itemrandomizerweb#23]
I've been playing around with Casual Mode a bit, and as someone trying to get into the randomizer. While it was a lot of fun, I got a lot of seeds that required fairly advanced tricks that require a significant amount of non-casual practice, that at least warrant some documentation that they are required to learn because it can be very confusing and frustrating to not really know when you need to backtrack vs when the seed just demands you do something advanced. Plowing through was great for learning some of these tricks, but it was several hours that I wasn't expecting for a "casual" playthrough and I ended up just giving up on about 2 of the first 5 seeds. Here's the advanced tricks I've seen required and (to my understanding) the constraint definitions that allow them:

- Mock Ball
Enabled by various powerups that normally require Speed Booster having the clause "(haveItem items Morph || haveItem items SpeedBooster)"
To remove: Remove the Morph part of the clause. For "Ice Beam", a clause requiring Speed Booster should be added.

- Ice clipping to Shaktool
Enabled by "Spring Ball" clause "(haveItem items Ice || haveItem items Grapple)"
To remove: Remove the Ice part of the clause.

- Hard climbs of the red tower
This is enabled by both a lack of protections from getting stuck in the red tower half of Brinstar due to not having Wave Beam to return (door glitching is a hard trick), and very low requirements for the Wrecked Ship (only power bombs and supers), which means a player can be forced to climb red tower almost vanilla.
To remove: Add additional constants to Wrecked Ship access, such as Space Jump || (Speed Booster && (Ice Beam || Wave Beam)). Disallow Wave Beam to spawn in locations before the Brinstar Wave Beam gate so that access to the surface is guaranteed with items only from the right side of the gate.

- Nasty hell runs
The energy requirements for some of the hell runs are pretty low. For example, Ice Beam only requires 2 tanks, which would be very tight even for an expert. I would suggest upping those requirements for casual. I didn't even figure in casual I'd be having to make hell runs, but if you do they should have a higher energy requirement that makes them reasonable, and less likely to be stuck with an impossible hell run if you don't have the tools to go back on the left side of the wave beam gate as above.

- No-movement-tools Xray crossing
Hell runs tend to get doubly-screwed because they stack those Etanks early for energy, and Xray has energy-only requirements which mean a hell run seed likes to put something important in there since it meets the energy requirements. You can end up having to cross it without even HJB. The energy-only requirements should at very least be tied to HJB for casual, if not removed entirely.

**Comment by Jaylow7** [comment]
100% agree with all of this. The point of a "Casual" setting should be to make the randomizer accessible to less skilled players. It should **not** require advanced speedrun tricks to complete. Out of nearly 30 seeds I've rolled on the current version I've been able to complete FOUR.

**Comment by tewtal** [comment]
Thanks for this information. I'm aware that the Casual mode is probably not what could be considered Causal right now and this will help a lot towards tuning it down so that it's more accessible for newer players. It's something I've wanted to do for a while but haven't had much time to look at.

**Comment by elreymu** [comment]
I agree with it, the only thing i would worry it's if runs would get similar too each other, i hope not though.

**Comment by theonlydude** [comment]
Hello mkfanha,
To know if you can finish a casual randomized rom with the techniques you know and get the list of all the possible techniques you can use the randomized Super Metroid roms solver (http://randommetroidsolver.pythonanywhere.com/).
